sound/soc/sof/intel/hda-dsp.c

```diff
@@ -630,8 +630,8 @@ static int hda_suspend(struct snd_sof_dev *sdev, bool runtime_suspend)
 	snd_hdac_ext_bus_link_power_down_all(bus);
 #endif
 
-	/* power down DSP */
-	ret = snd_sof_dsp_core_power_down(sdev, chip->host_managed_cores_mask);
+	/* power down all DSP cores */
+	ret = snd_sof_dsp_core_power_down(sdev, GENMASK(chip->cores_num - 1, 0));
```

Comment:
This probably warrants more cleanup -- we've had multiple issues stemming from unclarity on the split between DSP ops and core. And this seems one more in this category.

But this specific fix is good. I'd reword the first paragraph of the commit a bit:
""The 'host_managed_cores' field is a Intel HDA specific concept and it should not be used when calling generic SOF core functions like snd_sof_dsp_core_power_down. By mixing the two, the SOF core view of the DSP core power status can gets out of sync.""

I'm still a bit puzzled why the core_power_down call need to be in Intel specific suspend call. Isn't this something that should be common to all SOF implementations? I see we only enable cores in topology.c -- this is a bit weird, right. @lyakh @ranj063 ?

I wouldn't block this fix though. Given current interface split between core and the DSP ops, this fix seems both correct and required.

Reply:
It is a bit confusing here indeed I find. E.g. `hda_dsp_core_reset_power_down()` will only power down host-managed cores indeed. So, the comment "power down all cores" really isn't true. At least we should explain here in the comment, that we only have to update the enabled cores' mask. Otherwise I'm not sure we really need to send an additional IPC to power down those cores, not managed by the host. Isn't `SOF_IPC_PM_CTX_SAVE` a sufficient indication to the firmware, that we're about to shut it down?